app/jobs/calendar_importer/parsers/ics.rb

```diff
@@ -57,9 +56,12 @@ def import_events_from(data)
     end
 
     def parse_remote_calendars(data)
+      raise InvalidResponse, 'Source returned empty ICS data' if data.blank?
+
       Icalendar::Calendar.parse data
-    rescue StandardError => e
-      raise BadFeedResponse
+    rescue RuntimeError => e
+      # I hope this isn't swallowing up any important exceptions
```

Comment:
How could we test or investigate that?

Reply:
i attempted to test this [here](https://github.com/geeksforsocialchange/PlaceCal/blob/ik-1648-calendar-bad-source-state/test/jobs/calendar_importer/event_parse_ics_test.rb).

but ultimately the icalendar gem appears to lack a distinct exception class.

as for investigating- i think you'd have to trace the code path manually and see how the ICS parser tests for errors. so if there is a problem it will likely be in that code (or our input into it, which is tested on the line above). at least it is not catching `StandardError`.

or we could fork the icalendar gem and give it proper error reporting.

Reply:
maybe good to put all this in the comment for later?